### PR TITLE
Changing CLI for azure-consumption reflecting the new api-version

### PR DESCRIPTION
**Pull request by sandeepnl**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

### Command Guidelines

- [x] Each command and parameter has a meaningful description.
- [x] Each new command has a test.

(see [Authoring Command Modules](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules))

### CODE REVIEW

**Comment by sandeepnl** [comment]
@derekbekoe & @lmazuel Please review

File: src/command_modules/azure-cli-consumption/HISTORY.rst [hunk 1]
```diff
@@ -3,6 +3,10 @@
 Release History
 ===============
 
+0.2.0
+++++++++++
```
Comment (derekbekoe) [inline src/command_modules/azure-cli-consumption/HISTORY.rst]: Heading length is too long. It should match the no. of characters in the heading.
CI may fail due to this.

File: src/command_modules/azure-cli-consumption/HISTORY.rst [hunk 2]
```diff
@@ -3,6 +3,10 @@
...
+Release of new GA api version 2017-11-30.
```
Comment (derekbekoe) [inline src/command_modules/azure-cli-consumption/HISTORY.rst]: Should start with `* ` so it's an actual bullet point list.

File: src/command_modules/azure-cli-consumption/azure/cli/command_modules/consumption/_params.py [hunk 3]
```diff
@@ -10,7 +10,6 @@
...
 register_cli_argument('consumption usage list', 'include_additional_properties', options_list=('--include-additional-properties', '-a'), action='store_true', help='include additional properties in the usages')
 register_cli_argument('consumption usage list', 'include_meter_details', options_list=('--include-meter-details', '-m'), action='store_true', help='include meter details in the usages')
-register_cli_argument('consumption usage list', 'start_date', options_list=('--start-date', '-s'), type=get_datetime_type(), help='start date (in UTC Y-m-d) of the usages')
-register_cli_argument('consumption usage list', 'end_date', options_list=('--end-date', '-e'), type=get_datetime_type(), help='end date (in UTC Y-m-d) of the usages')
-register_cli_argument('consumption usage list', 'invoice_name', options_list=('--invoice-name', '-i'), help='name of a specific invoice to get the usage details that associate with')
+register_cli_argument('consumption usage list', 'start_date', options_list=('--start-date', '-s'), type=get_datetime_type(), help='start date (in UTC Y-m-d) of the usages. Both start date and end date need to be supplied or neither ')
```
Comment (derekbekoe) [inline src/command_modules/azure-cli-consumption/azure/cli/command_modules/consumption/_params.py]: nit: Remove the extra space at the end of the help text.

File: src/command_modules/azure-cli-consumption/azure/cli/command_modules/consumption/custom.py [hunk 4]
```diff
@@ -5,42 +5,35 @@
...
-        filter_expression = filter_to
-    if filter_from and filter_to:
+    if start_date and end_date:
+        filter_from = "properties/usageEnd ge \'{}\'".format(start_date.strftime("%Y-%m-%dT%H:%M:%SZ"))
+        filter_to = "properties/usageEnd le \'{}\'".format(end_date.strftime("%Y-%m-%dT%H:%M:%SZ"))
         filter_expression = "{} and {}".format(filter_from, filter_to)
-
-    return list(client.list(scope, expand=expand, filter=filter_expression, top=top))
+    if (start_date and not end_date) or (end_date and not start_date):
+        raise ValueError("Usage : [--start-date <startDate> & --end-date <endDate>].")
```
Comment (derekbekoe) [inline src/command_modules/azure-cli-consumption/azure/cli/command_modules/consumption/custom.py]: This should be done at the validator stage like this:
`"usage error: Both --start-date and --end-date need to be supplied or neither."`
https://github.com/Azure/azure-cli/blob/dev/src/command_modules/azure-cli-extension/azure/cli/command_modules/extension/_params.py#L23

Comment (sandeepnl) [reply src/command_modules/azure-cli-consumption/azure/cli/command_modules/consumption/custom.py]: Sure, but that example didnt help me on how I can put a validator spanning two arguments. I will sync with you offline for this

**Comment by sandeepnl** [comment]
Thanks @derekbekoe for merging. How can we get this tagged to be released for the Dec 4th release of CLI?

**Comment by derekbekoe** [comment]
Everything in dev branch is automatically included in each release so you should be good!

**Pull request merged**